resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientConfiguration.java

```diff
@@ -57,7 +57,7 @@ public void setProperties(Map<String, Object> newProps)
       providerFactory.setProperties(newProps);
    }
 
-   protected ResteasyProviderFactory getProviderFactory()
+   public ResteasyProviderFactory getProviderFactory()

```

Comment:
subjective comment:

I guess that the only reason for this is adding HeaderDelegate, right?

```java
ResteasyProviderFactory factory = ((ClientConfiguration)client.getConfiguration()).getProviderFactory();
factory.addHeaderDelegate(HeaderDelegateClass.class, new HeaderDelegate());
```

^^^ this looks quite ... complicated, WDYT? What about to use just:
```java
((ClientConfiguration)client.getConfiguration()).addHeaderDelegate(HeaderDelegateClass.class, new HeaderDelegate());
```
?

Reply:
@marekkopecky, ClientConfiguration doesn't have an addHeaderDelegate() method. Are you saying that we should add one? 

In fact, I found it surprisingly hard to add HeaderDelegates, on both the client and server sides. Why isn't it classified as a provider, so that it can be registered easily? I don't know.

Reply:
@marekkopecky,

re: "I guess that the only reason for this is adding HeaderDelegate, right?"

That's probably true.

Reply:
> ClientConfiguration doesn't have an addHeaderDelegate() method. Are you saying that we should add one?

I don't say that we should add one. But I see these advantages of this approach:
* easier adding of HeaderDelegate to client
* we don't provide access to all ResteasyProviderFactory methods, but just to the one method, that is really needed (`addHeaderDelegate`)

I currently doesn't see any disadvantage of this approach. Do you see some? WDYT?

Reply:
Hi @marekkopecky,

re: "I currently doesn't see any disadvantage of this approach. Do you see some?"

I've created RESTEASY-2059 "Simplify registration of HeaderDelegates".

Reply:
@ronsigal Yes, we can deal with this later, but if we make getProviderFactory() method public now, it could be complicated to make it private later ...

Anyway, this is PR for master branch, so it is not important for current  WF ...

Reply:
@marekkopecky, I've reverted getProviderFactory() to protected and added getParamConverter() and addHeaderDelegate().